File: sbe/binance/spot_3_2/generated-python/spot_sbe/TrailingDeltaFilter.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class TrailingDeltaFilter:
    """SBE message: TrailingDeltaFilter."""

    TEMPLATE_ID = 13
    SCHEMA_ID = 3
    SCHEMA_VERSION = 2
    BLOCK_LENGTH = 32

    def __init__(self):
        self.filter_type = None
        self.min_trailing_above_delta = None
        self.max_trailing_above_delta = None
        self.min_trailing_below_delta = None
        self.max_trailing_below_delta = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.min_trailing_above_delta is not None:
            buffer.write(struct.pack('<q', self.min_trailing_above_delta))
        if self.max_trailing_above_delta is not None:
            buffer.write(struct.pack('<q', self.max_trailing_above_delta))
        if self.min_trailing_below_delta is not None:
            buffer.write(struct.pack('<q', self.min_trailing_below_delta))
        if self.max_trailing_below_delta is not None:
            buffer.write(struct.pack('<q', self.max_trailing_below_delta))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0

        self.min_trailing_above_delta = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.max_trailing_above_delta = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.min_trailing_below_delta = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.max_trailing_below_delta = struct.unpack_from('<q', data, pos)[0]
        pos += 8

        # Skip to end of block for forward compatibility
        pos = 32

File: sbe/binance/spot_3_1/generated-php/spot_sbe/TrailingDeltaFilter.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class TrailingDeltaFilter
{
    public const TEMPLATE_ID = 13;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 1;
    public const BLOCK_LENGTH = 32;

    public int|float|array|null $filterType = null;
    public int|float|array|null $minTrailingAboveDelta = null;
    public int|float|array|null $maxTrailingAboveDelta = null;
    public int|float|array|null $minTrailingBelowDelta = null;
    public int|float|array|null $maxTrailingBelowDelta = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->minTrailingAboveDelta !== null) {
            $buffer .= pack('q', $this->minTrailingAboveDelta);
        }
        if ($this->maxTrailingAboveDelta !== null) {
            $buffer .= pack('q', $this->maxTrailingAboveDelta);
        }
        if ($this->minTrailingBelowDelta !== null) {
            $buffer .= pack('q', $this->minTrailingBelowDelta);
        }
        if ($this->maxTrailingBelowDelta !== null) {
            $buffer .= pack('q', $this->maxTrailingBelowDelta);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->minTrailingAboveDelta = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->maxTrailingAboveDelta = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->minTrailingBelowDelta = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->maxTrailingBelowDelta = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;

        // Skip to end of block for forward compatibility
        $offset = 32;

    }
}
